src/main/java/org/opensearch/flowframework/workflow/RegisterAgentStep.java

```diff
@@ -240,7 +231,7 @@ private String getLlmModelId(Map<String, String> previousNodeInputs, Map<String,
 
     private LLMSpec getLLMSpec(String llmModelId, Map<String, String> llmParameters) {
         if (llmModelId == null) {
-            throw new IllegalArgumentException("model id for llm is null");
+            throw new FlowFrameworkException("model id for llm is null", RestStatus.BAD_REQUEST);
```

Comment:
Can we pass workflow ID and step ID to this method to include in the exception? (Can handle in a future PR)

Reply:
https://github.com/opensearch-project/flow-framework/issues/244